File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/modules/teachers/TeacherDao.java
package pl.szczodrzynski.edziennik.data.db.modules.teachers;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface TeacherDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void add(Teacher teacher);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void addAll(List<Teacher> teacherList);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void addAllIgnore(List<Teacher> teacherList);

    @Query("DELETE FROM teachers WHERE profileId = :profileId")
    void clear(int profileId);

    @Query("SELECT * FROM teachers WHERE profileId = :profileId AND teacherId = :id")
    LiveData<Teacher> getById(int profileId, long id);

    @Query("SELECT * FROM teachers WHERE profileId = :profileId AND teacherId = :id")
    Teacher getByIdNow(int profileId, long id);

    @Query("SELECT * FROM teachers WHERE profileId = :profileId AND teacherType <= 127 ORDER BY teacherName, teacherSurname ASC")
    LiveData<List<Teacher>> getAllTeachers(int profileId);

    @Query("SELECT * FROM teachers WHERE profileId = :profileId ORDER BY teacherName, teacherSurname ASC")
    List<Teacher> getAllNow(int profileId);

    @Query("UPDATE teachers SET teacherLoginId = :loginId WHERE profileId = :profileId AND teacherId = :id")
    void updateLoginId(int profileId, long id, String loginId);
}

File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/dao/TeacherDao.kt
package pl.szczodrzynski.edziennik.data.db.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import androidx.sqlite.db.SupportSQLiteQuery
import pl.szczodrzynski.edziennik.data.db.entity.Teacher

@Dao
interface TeacherDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun add(teacher: Teacher)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun addAll(teacherList: List<Teacher>)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun addAllIgnore(teacherList: List<Teacher>)

    @RawQuery
    fun query(query: SupportSQLiteQuery): Int

    @Query("DELETE FROM teachers WHERE profileId = :profileId")
    fun clear(profileId: Int)

    @Query("SELECT * FROM teachers WHERE profileId = :profileId AND teacherId = :id")
    fun getById(profileId: Int, id: Long): LiveData<Teacher?>

    @Query("SELECT * FROM teachers WHERE profileId = :profileId AND teacherId = :id")
    fun getByIdNow(profileId: Int, id: Long): Teacher?

    @Query("SELECT * FROM teachers WHERE profileId = :profileId AND teacherType <= 127 ORDER BY teacherName, teacherSurname ASC")
    fun getAllTeachers(profileId: Int): LiveData<List<Teacher>>

    @Query("SELECT * FROM teachers WHERE profileId = :profileId ORDER BY teacherName, teacherSurname ASC")
    fun getAllNow(profileId: Int): List<Teacher>

    @Query("UPDATE teachers SET teacherLoginId = :loginId WHERE profileId = :profileId AND teacherId = :id")
    fun updateLoginId(profileId: Int, id: Long, loginId: String?)
}
